app/javascript/components/public_health/AdvancedFilter.js

```diff
@@ -0,0 +1,683 @@
+          {filterOption?.type === 'date' && (
+            <Col className="py-0" md="3">
+              {this.renderDateOptions(dateOption, index)}
+            </Col>
+          )}
+          <Col className="py-0">
+            {filterOption?.type === 'boolean' && (
+              <ButtonGroup toggle>
+                <ToggleButton
+                  type="checkbox"
+                  variant="outline-primary"
+                  checked={value}
+                  value="1"
+                  onChange={() => {
+                    this.changeValue(index, !value);
+                  }}>
+                  TRUE
+                </ToggleButton>
+                <ToggleButton
+                  type="checkbox"
+                  variant="outline-primary"
+                  checked={!value}
+                  value="0"
+                  onChange={() => {
+                    this.changeValue(index, !value);
+                  }}>
+                  FALSE
+                </ToggleButton>
+              </ButtonGroup>
+            )}
+            {filterOption?.type === 'option' && (
+              <Form.Group className="py-0 my-0">
+                <Form.Control
+                  as="select"
+                  value={value}
+                  className="py-0 my-0"
+                  onChange={event => {
+                    this.changeValue(index, event.target.value);
+                  }}>
+                  {filterOption.options.map((option, op_index) => {
+                    return (
+                      <option key={index + 'opkeyop-f' + op_index} value={option}>
+                        {option}
+                      </option>
+                    );
+                  })}
+                </Form.Control>
+              </Form.Group>
+            )}
+            {filterOption?.type === 'date' && dateOption != 'within' && (
+              <Form.Group className="py-0 my-0">
+                <DateInput
+                  date={value}
+                  onChange={date => {
+                    this.changeValue(index, date);
+                  }}
+                  placement="bottom"
+                  customClass="form-control-md"
+                  minDate={'1900-01-01'}
+                  maxDate={moment()
+                    .add(30, 'days')
+                    .format('YYYY-MM-DD')}
+                />
+              </Form.Group>
+            )}
+            {filterOption?.type === 'date' && dateOption === 'within' && (
+              <Form.Group className="py-0 my-0">
+                <Row>
+                  <Col className="pr-0">
+                    <DateInput
+                      date={value.start}
+                      onChange={date => {
+                        this.changeValue(index, { start: date, end: value.end });
+                      }}
+                      placement="bottom"
+                      customClass="form-control-md"
+                      minDate={'1900-01-01'}
+                      maxDate={moment()
```

Comment:
Is there a reason we would need to limit the min/max date here since the user is just searching by these dates and not setting a monitoree's fields?

I can't think of a situation where users might want to put in a minDate before 01/01/1900 but for the max date, I'm thinking of maybe a situation where a user might want to create a saved filter with some field having a max date well past 30 days (maybe an example of that would be "monitorees enrolled in 2020", the user wouldn't be able to set the end date of the filter to be 12/31/2020 because it's more than 30 days from now)

Reply:
Agreed with your line of thinking - I think the limits are important either way, but I am going to up the allowed max value.